src/blob/core.rs

```diff
@@ -30,6 +30,31 @@ where
     validate_data_during_index_regen: bool,
 }
 
+pub(crate) struct WriteResult {
+    dirty_bytes: u64
+}
+
+impl WriteResult {
+    pub(crate) fn dirty_bytes(&self) -> u64 {
+        self.dirty_bytes
+    }
+}
+
+pub(crate) struct DeleteResult {
+    dirty_bytes: u64,
+    success: bool
```

Comment:
According to the code below, this field means that the record was present and therefore deleted. I wouldn't call the field like that in that case. The better name would be `deleted`